### PR TITLE
Feature/cicd

### PR DESCRIPTION
**Pull request by adamschnaare**

### CODE REVIEW

File: practices/cicd/CICD.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# CI/CD
+
+- **Continuous Integration (CI)**: Automated integration of code
+- **Continuous Deployment (CD)**: Automated deployment of software
```
Comment (andrewlarsen514) [inline practices/cicd/CICD.md:4]: @adamschnaare when I think CI/CD I usually think of the spectrum with 3 bench marks.

Continuous Integration - Automated integration of code
Continuous Delivery - Automated integration of code and system; aka we know we _could_ deploy after each commit
Continuous Deployment - Automated deployment of software; we take that next step from Continuous Delivery and actually put something into production on each commit.

The difference between the last two, IMO, requires a high level of maturity with your Logging, Monitoring, Alerting, and Profiling.

Do you think it's worth covering the difference in this?

File: practices/cicd/CICD.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+[What is Continuous Integration](https://www.youtube.com/watch?v=1er2cjUq1UI)
+
+## CD
+
+CD is the task of automating release and deployment of code to running environments. Common tasks/checks that happen at this stage are:
+
+- Testing (unit, integration, &/or end-to-end)
+- Linting (code format checks)
+- Building/compiling
+- Release versioning
```
Comment (andrewlarsen514) [inline practices/cicd/CICD.md:33]: @adamschnaare is it worth bringing up the integrating of a system here? Not saying every deploy needs to spin up the entire ecosystem, but I generally see at least some end to end style testing with core components of the system

Comment (adamschnaare) [reply practices/cicd/CICD.md:33]: I guess for the purposes of the doc I was leaving this system agnostic. Would that step into architecture thought? Maybe this doc expands and has ideal definitions of entire ci/cd flows for different tech setups/architectures?

File: practices/cicd/CICD.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+Proper CI/CD automates the manual stuff and ensures that code is merged and released correctly.
+- 🏎 Team velocity
+- 💻 Code accuracy
+- 🤓 Developer happiness 
+
+## CI
+
+CI is the task of regularly merging code changes. It allows for common tasks to be run to ensure code quality and integrity. Some common tasks/checks that happen at this stage are:
+
+- Testing (unit, integration, &/or end-to-end)
```
[the remark this answers is not in the document]
Comment (adamschnaare) [reply practices/cicd/CICD.md:17]: Yeah, that would be good to point out. I was considering the unit tests as `code coverage` tests, and linting as code quality checks.

1. What processes for code quality were you thinking?
2. What process do you use for security scanning?

**Pull request merged**